nodebuilder/share/cmd/share.go

```diff
@@ -0,0 +1,209 @@
+	Args:  cobra.NoArgs,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		client, err := cmdnode.ParseClientFromCtx(cmd.Context())
+		if err != nil {
+			return err
+		}
+		defer client.Close()
+
+		prob := client.Share.ProbabilityOfAvailability(cmd.Context())
+		return cmdnode.PrintOutput(prob, nil, nil)
+	},
+}
+
+var getSharesByNamespaceCmd = &cobra.Command{
+	Use:   "get-by-namespace [dah, namespace]",
+	Short: "Gets all shares from an EDS within the given namespace.",
+	Args:  cobra.ExactArgs(2),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		client, err := cmdnode.ParseClientFromCtx(cmd.Context())
+		if err != nil {
+			return err
+		}
+		defer client.Close()
+
+		raw, err := parseJSON(args[0])
+		if err != nil {
+			return err
+		}
+
+		root := da.MinDataAvailabilityHeader()
+		err = json.Unmarshal(raw, &root)
+		if err != nil {
+			return err
+		}
+
+		ns, err := cmdnode.ParseV0Namespace(args[1])
+		if err != nil {
+			return err
+		}
+
+		shares, err := client.Share.GetSharesByNamespace(cmd.Context(), &root, ns)
+		return cmdnode.PrintOutput(shares, err, nil)
+	},
+}
+
+var getShare = &cobra.Command{
+	Use:   "get-share [dah, row, col]",
+	Short: "Gets a Share by coordinates in EDS.",
+	Args:  cobra.ExactArgs(3),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		client, err := cmdnode.ParseClientFromCtx(cmd.Context())
+		if err != nil {
+			return err
+		}
+		defer client.Close()
+
+		raw, err := parseJSON(args[0])
+		if err != nil {
+			return err
+		}
+
+		root := da.MinDataAvailabilityHeader()
+		err = json.Unmarshal(raw, &root)
+		if err != nil {
+			return err
+		}
+
+		row, err := strconv.ParseInt(args[1], 10, 64)
+		if err != nil {
+			return err
+		}
+
+		col, err := strconv.ParseInt(args[2], 10, 64)
+		if err != nil {
+			return err
+		}
+
+		s, err := client.Share.GetShare(cmd.Context(), &root, int(row), int(col))
+
+		formatter := func(data interface{}) interface{} {
+			sh, ok := data.(share.Share)
+			if !ok {
+				return data
+			}
+
+			ns := hex.EncodeToString(share.GetNamespace(sh))
+
+			return struct {
+				Namespace string `json:"namespace"`
+				Data      []byte `json:"data"`
+			}{
+				Namespace: ns,
+				Data:      share.GetData(sh),
+			}
+		}
+		return cmdnode.PrintOutput(s, err, formatter)
+	},
+}
+
+var getEDS = &cobra.Command{
+	Use:   "get-eds [dah]",
+	Short: "Gets the full EDS identified by the given root",
+	Args:  cobra.ExactArgs(1),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		client, err := cmdnode.ParseClientFromCtx(cmd.Context())
+		if err != nil {
+			return err
+		}
+		defer client.Close()
+
+		raw, err := parseJSON(args[0])
+		if err != nil {
+			return err
+		}
+
+		root := da.MinDataAvailabilityHeader()
+		err = json.Unmarshal(raw, &root)
+		if err != nil {
+			return err
+		}
+
+		shares, err := client.Share.GetEDS(cmd.Context(), &root)
+		return cmdnode.PrintOutput(shares, err, nil)
+	},
+}
+
+func parseJSON(param string) (json.RawMessage, error) {
```

Comment:
don't love this sitting here...thoughts about moving it to cmd/util if we must have it?

Reply:
The main concern that I have is that `parseJSON` is used only in share/cmd

Reply:
i know! i agree with you, it's convenient only here, but also feels like it doesn't belong, lets leave it there, i'm good